modules/swagger-codegen/src/main/resources/dart/class.mustache

```diff
@@ -1,14 +1,54 @@
-@Entity()
 class {{classname}} {
   {{#vars}}{{#description}}/* {{{description}}} */{{/description}}
-  @Property(name: '{{baseName}}')
   {{{datatype}}} {{name}} = {{{defaultValue}}};
   {{#allowableValues}}{{#min}} // range from {{min}} to {{max}}{{/min}}//{{^min}}enum {{name}}Enum { {{#values}} {{.}}, {{/values}} };{{/min}}{{/allowableValues}}
   {{/vars}}
   {{classname}}();
 
   @override
-  String toString()  {
+  String toString() {
     return '{{classname}}[{{#vars}}{{name}}=${{name}}, {{/vars}}]';
   }
+
+  {{classname}}.fromJson(Map<String, dynamic> json) {
+    if (json == null) return;
+  {{#vars}}
+  {{#isDateTime}}
+    {{name}} = json['{{name}}'] == null ? null : DateTime.parse(json['{{name}}']);
+  {{/isDateTime}}
+  {{^isDateTime}}
+    {{name}} =
+    {{#complexType}}
+        {{#isListContainer}}{{complexType}}.listFromJson(json['{{name}}']){{/isListContainer}}{{^isListContainer}}
+        {{#isMapContainer}}new {{complexType}}.mapFromJson(json['{{name}}']){{/isMapContainer}}
```

Comment:
This leads to a problem because mapFromJson rightly is not a constructor.

Example code generated:

```dart
    strings =
        
        new ProductStrings.mapFromJson(json['strings'])
        
    ;
```
Problems detected:
```
  error • A value of type 'ProductStrings' can't be assigned to a variable of type 'Map<String, ProductStrings>' at lib/model/product_representation.dart:57:9 • invalid_assignment
  error • The class 'ProductStrings' doesn't have a constructor named 'mapFromJson' at lib/model/product_representation.dart:57:28 • new_with_undefined_constructor
```
`new` needs to be removed. We should also add a test that uses maps.

Reply:
Hi again, sorry, that `new` was wrong of course.. 

I fixed that, and also removed the line for string "serialization". How would you add a test for this in general? All the testing is based on the petstore.yml at the moment. Would be best, to extend that API to include an API-call including a `Map`, or did you have a simpler test in mind @ircecho ?

Reply:
petstore.yaml only covers the basic while [petstore-with-fake-endpoints-models-for-testing.yaml](https://github.com/swagger-api/swagger-codegen/blob/master/modules/swagger-codegen/src/test/resources/2_0/petstore-with-fake-endpoints-models-for-testing.yaml) contains a lot more edge cases.

You may consider updating `./bin/dart-petstore.sh` to use petstore-with-fake-endpoints-models-for-testing.yaml.

Reply:
@wing328 Thanks, for the hint, I could have found this by myself 😁 
I gave it a short try and saw, that there are a LOT of errors to fix. There are things like:
- Classes starting with numbers
- Other names not matching dart requirements
- Maps in maps don't work at the moment
- maybe many more..

I started this a bit, but would provide a further PR if I have the time to fix these.